apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -679,7 +679,7 @@ public void downloadBlob(String container, String name, File destination, Execut
 
       ListeningExecutorService listeningExecutor = MoreExecutors.listeningDecorator(executor);
       RandomAccessFile raf = null;
-      File tempFile = new File(destination.getName() + "." + UUID.randomUUID());
+      File tempFile = new File(destination + "." + UUID.randomUUID());
```

Comment:
`destination` need not be absolute -- we want to make the temporary file *relative* to `destination`, not whatever path the application launched from.